client_test.go

```diff
@@ -954,3 +954,73 @@ func marshalCAPublicKey(curve cert.Curve, pubkey []byte) []byte {
 		panic("unsupported curve")
 	}
 }
+
+func TestGetOidcPollCode(t *testing.T) {
+	t.Parallel()
+
+	useragent := "dnclientUnitTests/1.0.0 (not a real client)"
+	ts := dnapitest.NewServer(useragent)
+	client := NewClient(useragent, ts.URL)
+	// attempting to defer ts.Close() will trigger early due to parallel testing - use T.Cleanup instead
+	t.Cleanup(func() { ts.Close() })
+	const expectedCode = "123456"
+	ts.ExpectRequest(message.PreAuthEndpoint, http.StatusOK, func(req message.RequestWrapper) []byte {
+		return jsonMarshal(message.PreAuthResponse{Data: message.PreAuthData{PollToken: expectedCode, LoginURL: "https://example.com"}})
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	resp, err := client.EndpointPreauth(ctx)
+	require.NoError(t, err)
+	assert.NotNil(t, resp)
+	assert.Equal(t, expectedCode, resp.PollToken)
+	assert.Equal(t, "https://example.com", resp.LoginURL)
+	assert.Empty(t, ts.Errors())
+	assert.Equal(t, 0, ts.RequestsRemaining())
+
+	//unhappy path
+	ts.ExpectRequest(message.PreAuthEndpoint, http.StatusBadGateway, func(req message.RequestWrapper) []byte {
+		return jsonMarshal(message.PreAuthResponse{Data: message.PreAuthData{PollToken: expectedCode, LoginURL: "https://example.com"}})
+	})
+	resp, err = client.EndpointPreauth(ctx)
+	require.Error(t, err)
+	require.Nil(t, resp)
+	assert.Empty(t, ts.Errors())
+	assert.Equal(t, 0, ts.RequestsRemaining())
+}
+
+func TestDoOidcPoll(t *testing.T) {
+	t.Parallel()
+
+	useragent := "dnclientUnitTests/1.0.0 (not a real client)"
+	ts := dnapitest.NewServer(useragent)
+	client := NewClient(useragent, ts.URL)
+	// attempting to defer ts.Close() will trigger early due to parallel testing - use T.Cleanup instead
+	t.Cleanup(func() { ts.Close() })
+	const expectedCode = "123456"
+	ts.ExpectRequest(message.EndpointAuthPoll, http.StatusOK, func(req message.RequestWrapper) []byte {
+		return jsonMarshal(message.EndpointAuthPollResponse{Data: message.EndpointAuthPollData{
+			Status:         "something",
+			EnrollmentCode: "",
+		}})
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	resp, err := client.EndpointAuthPoll(ctx, expectedCode)
+	require.NoError(t, err)
+	assert.Equal(t, resp.Status, "something")
+	assert.Equal(t, resp.EnrollmentCode, "")
```

Comment:
I think it would be better to test that we actually get a value back from the server into the struct correctly. (i.e. "foobar" or something at minimum)

Reply:
good call, added